3rdparty/find_dependencies.cmake

```diff
@@ -1603,3 +1649,22 @@ endif()
 # Compactify list of external modules.
 # This must be called after all dependencies are processed.
 list(REMOVE_DUPLICATES Open3D_3RDPARTY_EXTERNAL_MODULES)
+
+# Target linking order matters. When linking a target, the link libraries and
+# include directories are set. If the order is wrong, it can cause missing
+# symbols or wrong header versions. Generally, we want to link custom libs
+# before system libs; the order among differnt libs can also matter.
+#
+# Current rules:
+# 1. 3rdparty_curl should be linked before 3rdparty_png.
+# 2. Link "FROM_SOURCE" before "FROM_SYSTEM" targets.
```

Comment:
FROM_CUSTOM (or change to DOWNLOADED) before FROM_SYSTEM

Reply:
Done.